pkg/operator/highcpuusagealertcontroller/performanceprofile.go

```diff
@@ -0,0 +1,66 @@
+package highcpuusagealertcontroller
+
+import (
+	"context"
+	"encoding/json"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/utils/cpuset"
+)
+
+// default and taken from the docs
+const defaultCoresNum = 8
+
+var performanceGroup = schema.GroupVersionResource{Group: "performance.openshift.io", Version: "v2", Resource: "performanceprofiles"}
+
+type performanceProfileList struct {
+	Items []performanceProfile `json:"items"`
+}
+
+type performanceProfile struct {
+	Spec struct {
+		CPU *struct {
+			Reserved *string `json:"reserved"`
+		} `json:"cpu"`
+		NodeSelector map[string]string `json:"nodeSelector"`
+	} `json:"spec"`
+}
+
+func controlPlaneCores(ctx context.Context, client dynamic.Interface) (int, error) {
+	obj, err := client.Resource(performanceGroup).List(ctx, metav1.ListOptions{})
```

Comment:
Okay, after talking with the team, we are okay with using a dynamic client. That also means we should list resources from a lister, not directly from the API. So, you will have to start an informer and plug it into this controller.